src/codecoverage/bot/code_coverage_bot/uploader.py

```diff
@@ -96,13 +96,40 @@ def check_codecov_job():
         return False
 
 
-def gcp_upload(path, data):
+def gcp(repository, revision, data):
     '''
-    Upload a data payload on Google Cloud Storage
+    Upload a grcov raw report on Google Cloud Storage
+    * Compress with gzip
+    * Upload on bucket using revision in name
+    * Copy as latest file
     '''
-    assert isinstance(path, str)
-    assert isinstance(data, (bytes, str))
+    assert isinstance(data, bytes)
+    bucket = gcp_bucket()
 
+    # Compress report
+    archive = gzip.compress(data)
+
+    # Upload archive
+    path = '{}/{}.json'.format(repository, revision)
+    blob = bucket.blob(path)
+    blob.upload_from_string(archive)
+
+    # Update headers
+    blob.content_type = 'application/json'
+    blob.content_encoding = 'gzip'
+    blob.patch()
+
+    logger.info('Uploaded {} on {}'.format(path, bucket))
+
+    # Copy as latest file in same bucket
+    bucket.copy_blob(blob, bucket, new_name='{}/latest.json'.format(repository))
```

Comment:
I wonder if we will actually need the `latest.json` file. The backend could store this kind of info (requesting from hgmo the latest build, caching the value of course to avoid requesting it too often)

Reply:
I'll remove it from the PR, let's see how we make it work in the backend. Worst case, it's easy to add again.